scripts/coverage.sh

```diff
@@ -1,136 +1,69 @@
 #!/usr/bin/env bash
 #
-# Runs all tests and collects code coverage
+# Run tests and collect code coverage
 #
-# Warning: this process is a little slow
+# Run all:
+#   $ ./script/coverage.sh
 #
+# Run for specific packages
+#   $ ./script/coverage.sh -p solana-account-decoder
+#   $ ./script/coverage.sh -p solana-account-decoder -p solana-accounts-db [-p ...]
 
-if ! command -v grcov; then
-  echo "Error: grcov not found.  Try |cargo install grcov|"
-  exit 1
-fi
-
-if [[ ! "$(grcov --version)" =~ 0.8.[0-9] ]]; then
-  echo Error: Required grcov version not installed
+set -e
+here=$(cd -- "$(dirname -- "${BASH_SOURCE[0]}")" &>/dev/null && pwd)
 
-  echo "Installed version: $(grcov --version)"
+# Check for grcov commands
+if ! command -v grcov >/dev/null 2>&1; then
+  echo "Error: grcov not found.  Try |cargo install grcov|"
   exit 1
 fi
 
-set -e
-cd "$(dirname "$0")/.."
-source ci/_
+# Use nightly as we have some nightly-only tests (frozen-abi)
+# shellcheck source=ci/rust-version.sh
+source "$here/../ci/rust-version.sh" nightly
 
-cargo="$(readlink -f "./cargo")"
+# Clean up
+rm -rf "$here/../target" "$here/../coverage"
+find "$here/.." -type f -name '*.prof*' -exec rm {} +
+find "$here/.." -type f -name '*lcov*' -exec rm {} +
 
-: "${CI_COMMIT:=local}"
-reportName="lcov-${CI_COMMIT:0:9}"
+export RUSTFLAGS="-C instrument-coverage $RUSTFLAGS"
+export LLVM_PROFILE_FILE="default-%p-%m.profraw"
```

Comment:
could you add some reference why this env var need to be set? maybe instructed by grcov?

Reply:
come to think of it, it looks like `LLVM_PROFILE_FILE` can contain dirs.. so how about this? so that we can avoid `find`-ing on the reporoot dir.. (imo, this is a bit intrusive):

`export LLVM_PROFILE_FILE=$here/../target/cov/default-%p-%m.profraw`

Reply:
[57f67c0](https://github.com/anza-xyz/agave/pull/1260/commits/57f67c082e0fc05be4723ca9deb5392e4dc81ce2)